### PR TITLE
Fixes bug 1242983 - Added a mapping for correlations.

### PR DESCRIPTION
**Pull request by adngdb**
This mapping will probably need to be tuned a bit, but at least now you have a foundation on which you can start building your correlations app.

### CODE REVIEW

File: socorro/external/es/correlations.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import json
+import os
+
+from configman import Namespace, RequiredConfig
+from configman.converters import class_converter
+
+
+DIRECTORY = os.path.dirname(os.path.abspath(__file__))
```
Comment (peterbe) [inline socorro/external/es/correlations.py]: This is a pretty bad name :)
Perhaps call it `HERE` or `THIS_DIRECTORY`.

File: socorro/external/es/mappings/correlations_index_settings.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        "correlations": {
+            "_all": {
+                "enabled": false
+            },
+            "_source": {
+                "compress": true
+            },
+            "properties": {
+                "product": {
+                    "type": "string",
```
Comment (peterbe) [inline socorro/external/es/mappings/correlations_index_settings.json]: Why isn't this `analyzer: keyword` too? Is it out realistic fear that we might have corrections for "Firefox" AND "firefox"?

File: socorro/external/es/mappings/correlations_index_settings.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                },
+                "version": {
+                    "type": "string",
+                    "analyzer": "keyword",
+                    "norms" : {
+                        "enabled" : false
+                    }
+                },
+                "platform": {
+                    "type": "string",
```
Comment (peterbe) [inline socorro/external/es/mappings/correlations_index_settings.json]: Again, why not `keyword`?

File: socorro/external/es/mappings/correlations_index_settings.json [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                    "norms" : {
+                        "enabled" : false
+                    }
+                },
+                "signature": {
+                    "type": "string",
+                    "index": "not_analyzed",
```
Comment (peterbe) [inline socorro/external/es/mappings/correlations_index_settings.json]: Pardon my ignorance but what does this do?

File: socorro/external/es/mappings/correlations_index_settings.json [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+                },
+                "notes": {
+                    "type": "string",
+                    "norms" : {
+                        "enabled" : false
+                    }
+                },
+                "date": {
+                    "type": "date",
+                    "format": "date_time_no_millis"
```
Comment (peterbe) [inline socorro/external/es/mappings/correlations_index_settings.json]: I don't even think we'll have a time.

**Comment by peterbe** [comment]
This is excellent! But I think I need a bit more handholding on how to "embed" and actually use this.

**Comment by peterbe** [comment]
Perhaps we can chat about this in more detail in our 1:1

File: socorro/external/es/mappings/correlations_index_settings.json [hunk 6]
```diff
@@ -0,0 +1,65 @@
+{
+    "mappings": {
+        "correlations": {
```
Comment (peterbe) [inline socorro/external/es/mappings/correlations_index_settings.json]: note-to-self, this is the "doctype".

File: socorro/unittest/external/es/test_correlations.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+        correlations = Correlations(config=self.config)
+        correlations.create_correlations_index(
+            datetime.datetime.utcnow().date()
+        )
+
+        expected_index = today.strftime(ES_CORRELATIONS_INDEX)
+        self.indices.append(expected_index)
+
+        ok_(self.index_client.exists(expected_index))
```
Comment (peterbe) [inline socorro/unittest/external/es/test_correlations.py]: Perhaps add another couple of lines like this:
```python
# We should be able to create the index again without errors
correlations.create_correlations_index(
     datetime.datetime.utcnow().date()
)
```

File: socorro/external/es/mappings/correlations_index_settings.json (hunk 5 above)
Comment (adngdb) [inline socorro/external/es/mappings/correlations_index_settings.json]: Yeah, right, this should just be a ``"date"``.

File: socorro/external/es/correlations.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    def __init__(self, config):
+        super(Correlations, self).__init__()
+        self.config = config
+
+        self.indices_cache = set()
+
+    def get_index_for_date(self, date):
+        return date.strftime(self.config.elasticsearch_correlations_index)
+
+    def create_correlations_index(self, date):
```
Comment (peterbe) [inline socorro/external/es/correlations.py]: Can we change this to `(self, ex_index)` and remove like 47 instead. 
That way you can do this:
```python
idx = self.get_index_for_date(somedate)
self.create_correlation_index(idx)
self.connection.index(
    index=idx,
    doctype="correlations",
    ...
```

**Comment by adngdb** [comment]
@peterbe updated!

**Comment by peterbe** [comment]
r+

**Pull request merged**